mlrun/__main__.py

```diff
@@ -1091,6 +1102,17 @@ def project(
                     "token": proj.get_param("GIT_TOKEN"),
                 },
             )
+        if notification:
+            notifications = line2keylist(notification, keyname="type", valname="params")
+            for notification in notifications:
+                if notification["type"] == "file":
+                    with open(notification["params"]) as fp:
+                        lines = fp.read().splitlines()
+                        notification = list2dict(lines)
+                        add_notification2proj(notification, proj)
+
+                else:
+                    add_notification2proj({notification['type']:notification['params']}, proj)
```

Comment:
For testing, I'd suggest putting all this in a helper function.
That way you can test that function with notification inputs, and mock the `add_notification2proj` to make sure it gets the expected params.